### PR TITLE
Test springboot with checkpoint/retsore

### PR DESCRIPTION
**Pull request by anjumfatima90**

### CODE REVIEW

File: dev/com.ibm.ws.app.manager.springboot/src/com/ibm/ws/app/manager/springboot/util/SpringBootThinUtil.java [hunk 1]
```diff
@@ -318,6 +319,17 @@ protected String hash(JarFile jf, ZipEntry entry) throws IOException, NoSuchAlgo
         byte[] digested = digest.digest();
         return convertToHexString(digested);
     }
+    
+    private static MessageDigest getDigest(String... algorithms) throws NoSuchAlgorithmException{
+        for (String algorithm : algorithms) {
+            try {
+                return MessageDigest.getInstance(algorithm);
+            } catch (NoSuchAlgorithmException ignored) {
+                // Algorithm not available, try the next one. Allow auto FFDC here to capture the causing exception (if any).
```
Comment (tjwatson) [inline dev/com.ibm.ws.app.manager.springboot/src/com/ibm/ws/app/manager/springboot/util/SpringBootThinUtil.java]: I don't think you should FFDC here.  Instead save off the exception like this:

```java
    @FFDCIgnore(NoSuchAlgorithmException.class)
    private static MessageDigest getDigest(String... algorithms) throws NoSuchAlgorithmException {
        NoSuchAlgorithmException error = null;
        for (String algorithm : algorithms) {
            try {
                return MessageDigest.getInstance(algorithm);
            } catch (NoSuchAlgorithmException suppressed) {
                // Algorithm not available, save the error and try the next one.
                if (error != null) {
                    error.addSuppressed(suppressed);
                } else {
                    error = suppressed;
                }

            }
        }
        throw error;
    }
```

File: dev/io.openliberty.checkpoint.springboot.crac.app/.project [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>io.openliberty.checkpoint.springboot.crac.app</name>
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint.springboot.crac.app/.project]: Using the naming of of other spring boot test applications I suggest we name this `io.openliberty.checkpoint.springboot.fat30.app`

**Comment by anjumfatima90** [comment]
#build

File: dev/io.openliberty.checkpoint_fat_springboot/.classpath [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="fat/src"/>
+
+	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8">
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat_springboot/.classpath]: The test application requires Java 17, but the FAT test itself requires Java 8?

File: dev/io.openliberty.checkpoint_fat_springboot/bnd.bnd [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# 
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+javac.source: 17
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat_springboot/bnd.bnd:16]: This is not consistent with what is specified in the `.classpath` for Java 8.

File: dev/io.openliberty.checkpoint_fat_springboot/fat/src/io/openliberty/checkpoint/fat/springboot/BasicSpringBootTests.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipIfCheckpointNotSupported;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import componenttest.topology.utils.HttpUtils;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+
+@RunWith(FATRunner.class)
+@SkipIfCheckpointNotSupported
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat_springboot/fat/src/io/openliberty/checkpoint/fat/springboot/BasicSpringBootTests.java:34]: Is there a skip rule we need for requiring Java 17?

Comment (tjwatson) [reply dev/io.openliberty.checkpoint_fat_springboot/fat/src/io/openliberty/checkpoint/fat/springboot/BasicSpringBootTests.java:34]: `@MinimumJavaLevel(javaLevel = 17)` will be needed here.

File: dev/io.openliberty.checkpoint_fat_springboot/.classpath (hunk 3 above)
Comment (anjumfatima90) [reply dev/io.openliberty.checkpoint_fat_springboot/.classpath]: Done. Thanks

File: dev/io.openliberty.checkpoint_fat_springboot/fat/src/io/openliberty/checkpoint/fat/springboot/BasicSpringBootTests.java (hunk 5 above)
Comment (anjumfatima90) [reply dev/io.openliberty.checkpoint_fat_springboot/fat/src/io/openliberty/checkpoint/fat/springboot/BasicSpringBootTests.java:34]: Done. thanks!

**Comment by anjumfatima90** [comment]
#build

**Review by tjwatson — APPROVED** [review]

**Comment by anjumfatima90** [comment]
#libby

**Pull request merged**